optimum/habana/transformers/generation/utils.py

```diff
@@ -1469,7 +1469,7 @@ def sample(
             )
 
             # if eos_token was found in one sentence, set sentence to finished
-            if eos_token_id_tensor is not None:
+            if not lazy_mode and eos_token_id_tensor is not None:
```

Comment:
I think we should use `not ignore_eos` instead of `not lazy_mode`, so that it is consistent with [greedy search](https://github.com/huggingface/optimum-habana/blob/2cb7646bdaa41e88e05ef2bea18fce43f2faf0a8/optimum/habana/transformers/generation/utils.py#L1172) and that users can still do early stopping in lazy mode.

Reply:
yes, you are right